storage/client/queries/queries.go

```diff
@@ -680,16 +680,21 @@ const (
 			ref_swap_pairs.runtime = tokens.runtime AND
 			ref_swap_pairs.pair_address = ref_swap_pair_creations.pair_address
 		LEFT JOIN chain.evm_tokens AS ref_tokens ON
-		    ref_tokens.runtime = tokens.runtime AND
-		    ref_tokens.token_address = $5
+			ref_tokens.runtime = tokens.runtime AND
+			ref_tokens.token_address = $5
 		LEFT JOIN chain.evm_contracts as contracts ON (tokens.runtime = contracts.runtime AND tokens.token_address = contracts.contract_address)
 		WHERE
 			(tokens.runtime = $1) AND
 			($2::oasis_addr IS NULL OR tokens.token_address = $2::oasis_addr) AND
 			($3::text IS NULL OR tokens.token_name ILIKE '%' || $3 || '%' OR tokens.symbol ILIKE '%' || $3 || '%') AND
 			tokens.token_type IS NOT NULL AND -- exclude token _candidates_ that we haven't inspected yet
 			tokens.token_type != 0 -- exclude unknown-type tokens; they're often just contracts that emitted Transfer events but don't expose the token ticker, name, balance etc.
-		ORDER BY num_holders DESC, contract_addr
+		ORDER BY CASE
+			WHEN tokens.token_address = $4 THEN 1.0
+			WHEN ref_swap_pair_creations.token0_address = $4 AND ref_swap_pairs.reserve0 IS NOT NULL THEN ref_swap_pairs.reserve0::REAL / ref_swap_pairs.reserve1::REAL
+			WHEN ref_swap_pair_creations.token1_address = $4 AND ref_swap_pairs.reserve1 IS NOT NULL THEN ref_swap_pairs.reserve1::REAL / ref_swap_pairs.reserve0::REAL
+			ELSE 0.0
```

Comment:
Could you add a comment here explaining the sort order/calculation? If I'm reading it right, this sorts by the highest token price, not market cap? If that's the case I'd prefer to keep num_holders as the primary sort column since price could be manipulated.

Reply:
:+1: on adding comment. it is meant to be some cap though, it's multiplied by total supply lower. probably better to put these in named SELECT things. could also reduce some logic on the go side

Reply:
adding the comment

Reply:
oh ok nice, makes sense to me. thanks for the comment